clang/lib/CIR/CodeGen/CIRGenFunction.h

```diff
@@ -478,6 +478,13 @@ class CIRGenFunction : public CIRGenTypeCache {
   // applies to. nullptr if there is no 'musttail' on the current statement.
   const clang::CallExpr *MustTailCall = nullptr;
 
+  /// The attributes of cases collected during emitting the body of a switch
+  /// stmt.
+  llvm::SmallVector<llvm::SmallVector<mlir::Attribute, 4>, 2> caseAttrsStack;
+
+  /// The type of the condition for the emitting switch statement.
+  llvm::SmallVector<mlir::Type, 2> condTypeStack;
```

Comment:
This might have a problem of in face of nested switches: tracked globally at the function level. IIRC https://github.com/llvm/clangir/pull/528 tried to handle that by putting some of that information on the LexicalScope (where we have some tracking for switches), have you looked into that? An alternative would be to have a SwitchRAII thing that also updates, but effectively this is what LexicalScope does already.

Reply:
We have test for nested switch: https://github.com/llvm/clangir/blob/3b822725ea34933e2bf440436ed14ff28e7a13f5/clang/test/CIR/CodeGen/switch.cpp#L279-L287 . This is the reason why I make it a vector : 

RAII is good but given we only try to build switch in one function (buildSwitchStmt), I think we don't need to do that now. It won't be hard to do that we have multiple use places.